### PR TITLE
Fix/get open challenges

### LINKED ISSUE #2635: Fix get open challenges

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 1]
```diff
@@ -1163,9 +1163,10 @@ type ChallengesResponse struct {
 //	500:
 func (srh *StorageRestHandler) getOpenChallenges(w http.ResponseWriter, r *http.Request) {
 	var (
-		blobberID  = r.URL.Query().Get("blobber")
-		fromString = r.URL.Query().Get("from")
-		from       common.Timestamp
+		blobberID   = r.URL.Query().Get("blobber")
+		fromString  = r.URL.Query().Get("from")
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: remove from

File: code/go/0chain.net/smartcontract/dbs/event/challenge.go [hunk 2]
```diff
@@ -96,6 +113,28 @@ func (edb *EventDb) GetOpenChallengesForBlobber(blobberID string, from, now, cct
...
+	logging.Logger.Info("GetOpenChallengesForBlobber", zap.Any("result", result), zap.Any("chs", chs))
+
+	query = edb.Store.Get().Model(&Challenge{}).
+		Where("created_at = ? AND blobber_id = ? AND responded = 0", challengeWithChallengeID.CreatedAt, blobberID).
+		Limit(50).
+		Order(clause.OrderByColumn{
+			Column: clause.Column{Name: "created_at"},
+		}).
+		Order(clause.OrderByColumn{
+			Column: clause.Column{Name: "challenge_id"},
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/challenge.go]: you don't need 2 fields to sort

File: code/go/0chain.net/smartcontract/dbs/event/challenge.go [hunk 3]
```diff
@@ -69,25 +70,41 @@ func (edb *EventDb) GetChallenges(blobberId string, start, end int64) ([]Challen
...
 	}
 
+	logging.Logger.Info("GetOpenChallengesForBlobber", zap.Any("challengeWithChallengeID", challengeWithChallengeID), zap.Any("from", from), zap.Any("now", now), zap.Any("cct", cct), zap.Any("limit", limit), zap.Any("challengeID", challengeID))
+
 	query := edb.Store.Get().Model(&Challenge{}).
-		Where("created_at > ? AND blobber_id = ? AND responded = ?",
-			from, blobberID, 0).
-		Limit(limit.Limit).
-		Offset(limit.Offset).
+		Where("created_at > ? AND blobber_id = ? AND responded = 0", challengeWithChallengeID.CreatedAt, blobberID).
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/challenge.go]: use request I've provided

**Review by dabasov — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  | fix/get_open_challenges  |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  | fix/get_open_challenges  |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Pull request merged**